### PR TITLE
Make comment

### PR DESCRIPTION
**Pull request by jasonexus**
Made changes from fixed-get-comment pull request. I had a 400 bad request issue, so hopefully one of you are able to diagnose the issue for me. Thanks in advance!

### CODE REVIEW

File: src/components/comment-components/CommentList.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+function CommentList({allComments}: CommentListProps) {
+  return (
+    <div id="list-of-comments">
+        {allComments.length < 1 ? (
+          <div>
+            <h1>No comments yet!</h1>
+          </div>
+        ) : (
+          <ul>
+            {allComments.map((comment) =>
```
Comment (dellahumanita) [inline src/components/comment-components/CommentList.tsx]: I am getting an error here: `TypeError: allComments.map is not a function`

**Comment by dellahumanita** [comment]
The Comment button is not working for me, is this the expected behaviour?

**Review by dellahumanita — CHANGES_REQUESTED** [review]

File: src/components/forms/CommentForm.tsx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+  handleSubmit = (event: Event): void => {
+    var contentType = PLAIN;
+    if (this.state.markdown === true) {
+      var contentType = MARKDOWN;
+    }
+
+    let date = new Date().toISOString();
+
+    // const authorData = {
```
Comment (dalidia) [inline src/components/forms/CommentForm.tsx]: Nitpicky thing but can you remove this commented out code?

File: src/components/Post.tsx [hunk 3]
```diff
@@ -1,75 +1,201 @@
...
+  return (
+    <li className="bg-zinc-100 border-solid border-1 border-slate-600 w-2/3 m-auto rounded-lg py-4 px-5  my-5">
+      <div className="grid grid-cols-1 gap-y-2">
+        <div className="flex flex-row justify-between">
+          <span className="font-semibold tracking-wide text-lg">
+            {authorName}
+          </span>
+
+          {/* Display these buttons if the author of the  post is the current author */}
+          {authorName === currentUser && (
```
Comment (dalidia) [inline src/components/Post.tsx:113]: Not related to your code, but I assumed that this part was checking the id and not the name. Definitely something we want to change asap, again not in your code but in a different PR

Comment (dalidia) [reply src/components/Post.tsx:113]: Thanks for changing the variable names, that is helping

File: src/components/PostList.tsx [hunk 4]
```diff
@@ -22,13 +22,14 @@ function PostList({ initialPosts, currentAuthor, onRemove } : PostListProps){
...
                     .map(post => (
                         <Post
-                            id={post.id}
-                            title={post.title}
-                            body={post.description}
-                            author={post.author} 
-                            currentAuthor={currentAuthor}
-                            onRemove={onRemove}
-                            contentType={post.contentType}
+                        postId={post.postId}
```
Comment (dalidia) [inline src/components/PostList.tsx]: Nitpicky thing, but can you fix the indentation for <Post >

**Review by dalidia — CHANGES_REQUESTED** [review]
Have you merged master? I checked the db and you're not using the latest changes for the backend in which we don't use numbers as ID anymore

File: src/components/Post.tsx [hunk 5]
```diff
@@ -91,14 +92,20 @@ function Post({
...
     function fetchComments(authorId: string, postId: string) {
-      getAllComments(authorId.toString(), postId)
+      getAllComments(authorId, postId)
         .then((data) => setComments(data))
         .catch((err) => {
           setErrMsg(err.message);
         });
     }
 
+    function getLikesForPosts(authorId: string, postId: string){
```
Comment (dalidia) [inline src/components/Post.tsx]: don't add extra functionalities like likes for posts. We are checking comments rn

**Comment by jasonexus** [comment]
Added the model fixes from Dan and made a few adjustments afterwards. Getting comments and making comments should work properly now. Will try to make a few more adjustments since new comments only appear when you refresh the page.

**Review by dalidia — APPROVED** [review]
PR seems good! Thank you!

**Review by dellahumanita — APPROVED** [review]

**Pull request merged**